rclcpp/include/rclcpp/parameter.hpp

```diff
@@ -117,50 +121,62 @@ class Parameter
   decltype(auto)
   get_value() const;
 
+  /// Get value of parameter as boolean.
```

Comment:
Do these throw if they are the wrong type? I would add that in here. Same for all the as_* methods below.